src/main/java/appeng/core/sync/packets/PacketClick.java

```diff
@@ -84,6 +85,18 @@ public void serverPacketData(final INetworkInfo manager, final AppEngPacket pack
                         this.hitX,
                         this.hitY,
                         this.hitZ);
+            } else if (is.getItem() instanceof ToolAdvancedNetworkTool atnt) {
+                atnt.serverSideToolLogic(
```

Comment:
Instead of duplication you can create new interface `HasServerSideToolLogic` that have only method and check it here